cmd/tx_lookup.cpp

```diff
@@ -0,0 +1,120 @@
+    fs::create_directories(etl_path);
+    etl::Collector collector(etl_path.string().c_str(), /* flush size */ 512 * kMebi);
+
+    lmdb::DatabaseConfig db_config{db_path};
+    db_config.set_readonly(false); 
+    std::shared_ptr<lmdb::Environment> env{lmdb::get_env(db_config)};
+    std::unique_ptr<lmdb::Transaction> txn{env->begin_rw_transaction()};
+
+    auto bodies_table{txn->open(db::table::kBlockBodies)};
+    auto tx_lookup_table{txn->open(db::table::kTxLookup)};
+    auto transactions_table{txn->open(db::table::kEthTx)};
+
+    try {
+        auto initial_block_number{db::stages::get_stage_progress(*txn, db::stages::kTxLookupKey)};
+        // Extract
+        Bytes start(8, '\0');
+        boost::endian::store_big_u64(&start[0], initial_block_number);
+        uint64_t current_block_number{initial_block_number};
+        MDB_val key_mdb{db::to_mdb_val(start)};
+        MDB_val data_mdb;
+        SILKWORM_LOG(LogInfo) << "Started Transaction Lookup Extraction" << std::endl;
+        uint64_t total_entries = 0;
+        for (int rc{bodies_table->seek(&key_mdb, &data_mdb)}; rc != MDB_NOTFOUND; rc = bodies_table->get_next(&key_mdb, &data_mdb)) {
+            auto body_rlp{db::from_mdb_val(data_mdb)};
+            auto body{db::detail::decode_stored_block_body(body_rlp)};
+            Bytes block_number(static_cast<unsigned char*>(key_mdb.mv_data), 8);
+
+            if (body.txn_count == 0) {
+                current_block_number++;
+                continue;
+            }
+            Bytes transaction_key(8, '\0');
+            boost::endian::store_big_u64(transaction_key.data(), body.base_txn_id);
+            MDB_val tx_key_mdb{db::to_mdb_val(transaction_key)};
+            MDB_val tx_data_mdb{};
+
+            uint64_t i{0};
+            for (int rc{transactions_table->seek_exact(&tx_key_mdb, &tx_data_mdb)}; rc != MDB_NOTFOUND && i < body.txn_count; rc = transactions_table->get_next(&tx_key_mdb, &tx_data_mdb), ++i) {
+                // Take transaction rlp, then hash it in order to get the transaction hash
+                ByteView tx_rlp{db::from_mdb_val(tx_data_mdb)};
+                auto hash{keccak256(tx_rlp)};
+                total_entries++;
+                etl::Entry entry{Bytes(hash.bytes, 32), block_number};
+                collector.collect(entry);
+            }
+            current_block_number++;
+        }
+        SILKWORM_LOG(LogInfo) << "Entries to be inserted << " << total_entries << std::endl;
+        SILKWORM_LOG(LogInfo) << "Started Transaction Lookup Loading" << std::endl;
+        // If it was not empty before appending cannot happen
+        if (initial_block_number == 0) {
+            // Clear lookups just to be sure there is no garbage when append is enabled
+            txn->open(db::table::kTxLookup)->clear();
+            collector.load(tx_lookup_table.get(), nullptr, MDB_APPEND);
+        } else {
+            collector.load(tx_lookup_table.get(), nullptr, 0);
+        }
```

Comment:
MDB_APPEND is way faster when bulk loading.
I'd suggest the "else" case to delete all records from initial_block_number and eventually load collector data wiyh MDB_APPEND

Reply:
We are talking about hashes in key, we are not sure whether they are already sorted accordingly to the previous ones